MerlinAU.sh

```diff
@@ -6375,7 +6429,23 @@ _PostUpdateEmailNotification_()
          [ "$(nvram get ntp_ready)" -eq 1 ] && \
          [ "$(nvram get start_service_ready)" -eq 1 ] && \
          [ "$(nvram get success_start_service)" -eq 1 ]
-      then break ; fi
+      then
+         # Only check for unbound if checkUnbound is true
+         if $checkUnbound; then
+            # Check if unbound service is running
+            if ps | grep -w "unbound" | grep -v "grep" > /dev/null; then
+               Say "All required services, including unbound, are up and running."
+               break
+            else
+               Say "Unbound service not running yet. Waiting..."
+            fi
+         else
+            Say "All required services are up and running (unbound check skipped)."
+            break
+         fi
+      else
+         Say "System services not fully ready. Waiting..."
```

Comment:
Check for Unbound service running if installed? Feedback? 
Suggestions on this idea?

Reply:
> Check for Unbound service running if installed? Feedback? Suggestions on this idea?
>
I don't understand the reasoning for checking Unbound before sending the post-reboot update email notification.
What's the benefit of doing that? What's the link between Unbound and email notifications?

Many users, including myself, don't even have Unbound; and there are some people who say they have it installed but it's disabled/inactive (for some reason).

Reply:
> > Check for Unbound service running if installed? Feedback? Suggestions on this idea?
> 
> I don't understand the reasoning for checking Unbound before sending the post-reboot update email notification. What's the benefit of doing that? What's the link between Unbound and email notifications?

To send an email is it using an IP address or a DNS name? If it's using DNS then it's using unbound is my understanding (if installed). On my router I did a test where I killed the unbound service and all network access stopped. And I remember a user mentioning that when unbound stopped their network access stopped as well.

Maybe that's a configuration choice for users here?:

![image](https://github.com/user-attachments/assets/900c6554-fe9b-4f94-bd7f-7acf354029c5)

> Many users, including myself, don't even have Unbound; and there are some people who say they have it installed but it's disabled/inactive (for some reason).

This is why I made the check conditional on it being installed, but I'm surprised that you know users that have it installed and disabled, because when I killed the service on my router my network access completely died.

Not sure how your email functions completely work so that's why I'm asking for opinions if waiting for DNS to be fully up is helpful?

Reply:
@Martinski4GitHub 

I just retested and confirmed, if I shutdown unbound properly from the unbound manager addon; my internet still works. But if I force kill the service, my network access is in limbo until I restart the service:

![image](https://github.com/user-attachments/assets/486135f0-df34-499d-bf1c-e1431d066c16)

Reply:
> > > Check for Unbound service running if installed? Feedback? Suggestions on this idea?
> > 
> > 
> > I don't understand the reasoning for checking Unbound before sending the post-reboot update email notification. What's the benefit of doing that? What's the link between Unbound and email notifications?
> 
> To send an email is it using an IP address or a DNS name? If it's using DNS then it's using unbound is my understanding (if installed). On my router I did a test where I killed the unbound service and all network access stopped. And I remember a user mentioning that when unbound stopped their network access stopped as well.
> 
> Maybe that's a configuration choice for users here?:
> ...
> 
> > Many users, including myself, don't even have Unbound; and there are some people who say they have it installed but it's disabled/inactive (for some reason).
> 
> This is why I made the check conditional on it being installed, but I'm surprised that you know users that have it installed and disabled, because when I killed the service on my router my network access completely died.
> 
> Not sure how your email functions completely work so that's why I'm asking for opinions if waiting for DNS to be fully up is helpful?
>
>
Yes, for email notifications to work, the DNS service (e.g. **dnsmasq**) must be working because it must resolve the address of the SMTP mail server (e.g. smtp.gmail.com). Once the WAN interface is in a connected state and active, critical built-in services like DNS are started so the NTP daemon can sync the system clock, and the other non-critical built-in services (e.g. FTP, SSH, SMB, VPN, etc.) get started. DNS must be working at this point because the NTP daemon must also be able to resolve the address of the NTP server (e.g. time.nist.gov) in order to sync the system clock.

Now, I've never used Unbound and I have no idea how it works, so I can only assume that before the Unbound service is started, the built-in DNS service (**dnsmasq**) is active and able to resolve any URL address. If that's not the case and everything is "on hold" until Unbound is started, it would indicate a serious flaw in the design of Unbound. Given its popularity, I'm assuming that Unbound can happily co-exist with **dnsmasq** until it gets started to take over the DNS functions. But again, I don't know exactly what happens during reboot when Unbound is involved.

The main reason I don't install 3rd-party add-ons that attempt to replace existing built-in services that are critical for network connectivity (e.g. DNS, NTP, firewall) is that these scripts have not been fully integrated with the embedded system to work seamlessly and take over functions transparently without affecting other components that rely on critical network connectivity.

Reply:
> @Martinski4GitHub
> 
> I just retested and confirmed, if I shutdown unbound properly from the unbound manager addon; my internet still works. 
>
That's probably how some users say they have it "installed but disabled." The same thing can be done with Diversion: you can have it installed and yet set to disabled.

>But if I force kill the service, my network access is in limbo until I restart the service:
> 
> 
> To me this indicated the unbound service was somewhat important if installed; and should probably be running to have external network access (as long as it's DNS traffic)
>
>
See my previous response. IOW, if **dnsmasq** is up  & running during the boot sequence **before** Unbound gets started, we have DNS resolution.

Reply:
> > > > Check for Unbound service running if installed? Feedback? Suggestions on this idea?
> > > 
> > > 
> > > I don't understand the reasoning for checking Unbound before sending the post-reboot update email notification. What's the benefit of doing that? What's the link between Unbound and email notifications?
> > 
> > 
> > To send an email is it using an IP address or a DNS name? If it's using DNS then it's using unbound is my understanding (if installed). On my router I did a test where I killed the unbound service and all network access stopped. And I remember a user mentioning that when unbound stopped their network access stopped as well.
> > Maybe that's a configuration choice for users here?:
> > ...
> > > Many users, including myself, don't even have Unbound; and there are some people who say they have it installed but it's disabled/inactive (for some reason).
> > 
> > 
> > This is why I made the check conditional on it being installed, but I'm surprised that you know users that have it installed and disabled, because when I killed the service on my router my network access completely died.
> > Not sure how your email functions completely work so that's why I'm asking for opinions if waiting for DNS to be fully up is helpful?
> 
> Yes, for email notifications to work, the DNS service (e.g. **dnsmasq**) must be working because it must resolve the address of the SMTP mail server (e.g. smtp.gmail.com). Once the WAN interface is in a connected state and active, critical built-in services like DNS are started so the NTP daemon can sync the system clock, and the other non-critical built-in services (e.g. FTP, SSH, SMB, VPN, etc.) get started. DNS must be working at this point because the NTP daemon must also be able to resolve the address of the NTP server (e.g. time.nist.gov) in order to sync the system clock.
> 
> Now, I've never used Unbound and I have no idea how it works, so I can only assume that before the Unbound service is started, the built-in DNS service (**dnsmasq**) is active and able to resolve any URL address. If that's not the case and everything is "on hold" until Unbound is started, it would indicate a serious flaw in the design of Unbound. Given its popularity, I'm assuming that Unbound can happily co-exist with **dnsmasq** until it gets started to take over the DNS functions. But again, I don't know exactly what happens during reboot when Unbound is involved.
> 
> The main reason I don't install 3rd-party add-ons that attempt to replace existing built-in services that are critical for network connectivity (e.g. DNS, NTP, firewall) is that these scripts have not been fully integrated with the embedded system to work seamlessly and take over functions transparently without affecting other components that rely on critical network connectivity.

It's all good questions and points. 

By the time you responded I was counting sheep hahaha! 
I'll have some free time today though so I'll go through the motions of testing this shortly and advising :)

Reply:
I can confirm that the **dnsmasq** service is up at boot, and that waiting for Unbound doesn't appear to be a requirement. Also without this check in your last PR when I tested it ran fine which reconfirms this is not a requirement. :)

I was playing around with the idea when we were troubleshooting the email issue; because I noticed both me and Tom had Unbound installed.

Reply:
Removed!